homeassistant/components/netgear/sensor.py

```diff
@@ -119,3 +207,60 @@ def async_update_device(self) -> None:
         self._active = self._device["active"]
         if self._device.get(self._attribute) is not None:
             self._state = self._device[self._attribute]
+
+
+class NetgearRouterTrafficEntity(NetgearRouterEntity, SensorEntity):
+    """Representation of a device connected to a Netgear router."""
+
+    _attr_entity_registry_enabled_default = False
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        router: NetgearRouter,
+        attribute: str,
+    ) -> None:
+        """Initialize a Netgear device."""
+        super().__init__(coordinator, router)
+        self._attribute = attribute
+        self.entity_description = SENSOR_TRAFFIC_TYPES[self._attribute]
+        self._name = f"{router.device_name} {self.entity_description.name}"
+        self._unique_id = f"{router.serial_number}-{self._attribute}"
+        self._traffic = None
+        self._traffic_avg = None
+        self._traffic_avg_attr = f"{self.entity_description.name} average"
+        self.async_update_device()
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        return self._traffic
+
+    @property
+    def extra_state_attributes(self):
+        """Return device specific state attributes."""
+        extra_attr = {}
+        if self._traffic_avg is not None:
+            extra_attr[self._traffic_avg_attr] = self._traffic_avg
+        return extra_attr
+
+    @callback
+    def async_update_device(self) -> None:
+        """Update the Netgear device."""
+        if self._router.traffic_data is not None:
+            data = self._router.traffic_data.get(self._attribute)
+            if isinstance(data, float):
```

Comment:
Please set a lambda or function per description instead that accesses the correct coordinator data.

Example:
https://github.com/home-assistant/core/blob/b7007b364a133d2e0fad187559972ee4ebbb4c5c/homeassistant/components/airly/sensor.py#L100

Reply:
done